### PR TITLE
Fix isomorphisms for semigroups 

### PR DESCRIPTION
**Pull request by james-d-mitchell**
### Description of changes (for the release notes)

The overall purpose of this PR is:

1. Make `AsSemigroup` and `AsMonoid` operations rather than attributes, so that, for example, methods for things like `AsSemigroup(IsPartialPermSemigroup, S)` can be installed.
2. Fix several bugs in the methods `IsomorphismXSemigroup/Monoid` where `X` is one of `PartialPerm`, `Transformation`, and `Fp`. I also refactored some of this code to remove duplication, and some unnecessary methods.
3.  Fix some minor issues with `ViewString` and `String` methods for semigroups and monoids. 

I added tests which cover almost everything in `semitran.gi` and `semipperm.gi` and hence which cover all the issues resolved in this PR. I amalgamated the test files `semitrans.tst` and `semitran.tst` in `testinstall`.

The issues addressed in this pull request are:

* The wrong `ViewString` method was used for inverse semigroups without `GeneratorsOfInverseSemigroup`
* There were no methods for `String` for certain types of inverse semigroups and monoids
* There was no default method for `IsGeneratorsOfInverseSemigroup`
* There was no method for `ZeroMutable/Immutable` for a collection of partial perms, and the wrong method was used for `OneMutable/Immutable`
* `IsomorphismTransformationMonoid` and `IsomorphismPartialPermMonoid` sometimes did not return monoid isomorphisms, and their range sometimes was not a monoid.
* The inverse of `IsomorphismReesZeroMatrixSemigroup`was not defined on the zero of its range (so was not actually the inverse)
* The inverse of `IsomorphismFpSemigroup` when applied to an fp monoid was not properly defined (resulted in an error)
* The inverse of `IsomorphismTransformationSemigroup/Monoid` when applied to a semigroup/monoid of partial perms was not properly defined (the function was not the inverse). This also highlighted some issues with `AsPartialPerm`, which were also resolved.
* The method for `IsomorphismTransformationSemigroup/Monoid` for perm groups did not work
* The method for `NrFixedPoints` for a partial perm collection, returned the number of moved points
* The method for `MeetOfPartialPerms` did not work in some cases

### CODE REVIEW

**Comment by james-d-mitchell** [comment]
I think this is ready to go now. If this this branch is tested against Semigroups 2.8.0, then the tests will fail (only in the text of the output, no errors). I have prepared a new version (2.8.1) which I'll release when/if this is merged into master

File: lib/fpmon.gi [hunk 1]
```diff
@@ -210,7 +210,7 @@ function( F, rels )
     if Length(gens) > Length(rels) then
       SetIsFinite(s, false);
     fi;
-	
+
```
Comment (ChrisJefferson) [inline lib/fpmon.gi]: Is there any actual changes up to this point in this file? Looks like fiddling with whitespace.

Comment (james-d-mitchell) [inline lib/fpmon.gi]: I changed all the tabs in this file to spaces in a single commit, then fixed the function in a separate commit. So, no there are no actual changes up to this point.

**Comment by james-d-mitchell** [comment]
I guess I have to rework this to get it reviewed, so I'll try again later.

**Pull request closed without merging**